mlflow/projects/_project_spec.py

```diff
@@ -199,7 +200,8 @@ def _compute_path_value(self, user_param_value, storage_dir):
         basename = os.path.basename(user_param_value)
```

Comment:
@trangevi While we're here, I it just occurred to me that `os.path.basename(user_param_value)` may not always operate correctly on URIs. For example, invoking `os.path.basename()` on a POSIX-style path from a Windows machine likely does not produce the expected output. Further, multiple path parameter values with the same basename will collide under the current convention, causing unexpected behavior / failures (e.g.., `param1` with value `/path/to/small/model.txt`  and `param2 ` with value `/path/to/big/model.txt` both have the same base name: `model.txt`).

I realize that this code isn't directly modified by your change, but it would be great to fix up some of these longstanding path issues. Instead of this basename logic, perhaps we can use a parameter's index in the list of parameters as its artifact directory base name (e.g., `<storage_dir>/param_0` for parameter 0, `<storage_dir>/param_1` for parameter 1, ...). Let me know if this makes sense to you; I had thought about using the parameter name, but this may contain characters that produce unexpected behavior (i.e., the string `C:\\` in a parameter name might not play nicely with a Windows system).